controllers/backupcronjob/backupcronjob_controller.go

```diff
@@ -0,0 +1,366 @@
+func (r *BackupCronJobReconciler) startCron(ctx context.Context, backUpConfig *controllerv1alpha1.BackupCronJobConfig, logger logr.Logger) {
+	log := logger.WithName("backup cron")
+	log.Info("Starting backup cron scheduler")
+
+	// remove existing cronjob tasks
+	// we cannot update the existing tasks, so we need to remove them and add new ones
+	entries := r.cron.Entries()
+	for _, entry := range entries {
+		log.Info("Removing existing cronjob task", "entryID", entry.ID)
+		r.cron.Remove(entry.ID)
+	}
+
+	// add cronjob task
+	log.Info("Adding cronjob task", "schedule", backUpConfig.Schedule)
+	_, err := r.cron.AddFunc(backUpConfig.Schedule, func() {
+		taskLog := logger.WithName("cronTask")
+
+		taskLog.Info("Starting DevWorkspace backup job")
+		if err := r.executeBackupSync(ctx, logger); err != nil {
+			taskLog.Error(err, "Failed to execute backup job for DevWorkspaces")
+		}
+		taskLog.Info("DevWorkspace backup job finished")
+	})
+	if err != nil {
+		log.Error(err, "Failed to add cronjob function")
+		return
+	}
+
+	log.Info("Starting cron scheduler")
+	r.cron.Start()
+}
+
+// stopCron stops the cron scheduler and removes all existing cronjob tasks.
+func (r *BackupCronJobReconciler) stopCron(logger logr.Logger) {
+	log := logger.WithName("backup cron")
+	log.Info("Stopping cron scheduler")
+
+	// remove existing cronjob tasks
+	entries := r.cron.Entries()
+	for _, entry := range entries {
+		r.cron.Remove(entry.ID)
+	}
+
+	ctx := r.cron.Stop()
+	ctx.Done()
+
+	log.Info("Cron scheduler stopped")
+}
+
+// executeBackupSync executes the backup job for all DevWorkspaces in the cluster that
+// have been stopped in the last N minutes.
+func (r *BackupCronJobReconciler) executeBackupSync(ctx context.Context, logger logr.Logger) error {
+	log := logger.WithName("executeBackupSync")
+	log.Info("Executing backup sync for all DevWorkspaces")
+	devWorkspaces := &dw.DevWorkspaceList{}
+	err := r.List(ctx, devWorkspaces)
+	if err != nil {
+		log.Error(err, "Failed to list DevWorkspaces")
+		return err
+	}
+	for _, dw := range devWorkspaces.Items {
+		if !r.wasStoppedInTimeRange(&dw, 30, ctx, logger) {
+			log.Info("Skipping backup for DevWorkspace that wasn't stopped recently", "namespace", dw.Namespace, "name", dw.Name)
+			continue
+		}
+		dwID := dw.Status.DevWorkspaceId
+		log.Info("Found DevWorkspace", "namespace", dw.Namespace, "devworkspace", dw.Name, "id", dwID)
+
+		if err := r.createBackupJob(&dw, ctx, logger); err != nil {
+			log.Error(err, "Failed to create backup Job for DevWorkspace", "id", dwID)
+			continue
+		}
+		log.Info("Backup Job created for DevWorkspace", "id", dwID)
+
+	}
+	return nil
+}
+
+// wasStoppedInTimeRange checks if the DevWorkspace was stopped in the last N minutes.
+func (r *BackupCronJobReconciler) wasStoppedInTimeRange(workspace *dw.DevWorkspace, timeRangeInMinute float64, ctx context.Context, logger logr.Logger) bool {
```

Comment:
nit, `ctx` seems unused here:
```suggestion
func (r *BackupCronJobReconciler) wasStoppedInTimeRange(workspace *dw.DevWorkspace, timeRangeInMinute float64, logger logr.Logger) bool {
```